REL: 3.4.0 Release

sebix opened the pull request:
this commit contains the entries in changelog and NEWS
updates the linux package documentation
adds upgrade scripts for removed/renamed bots
and sets the version number

I decided that the version number 3.4.0 fits better as a lot of changes we had are much more than bug fixes

aaronkaplan commented on CHANGELOG.md:
```diff
@@ -30,8 +26,6 @@
   - Log the downloaded size in bytes to ease troubleshooting (PR#2554 by Sebastian Wagner).
   - Fix import for Timeout exception preventing another exception (fixes #2555, PR#2556 by Sebastian Wagner).
 - Remove `intelmq.bots.collectors.twitter` as it uses an unmaintained library and does not work any more (fixes #2346, #2441, PR#2568 by Sebastian Wagner).
-- Renamed `intelmq.bots.parser.twitter` to `intelmq.bots.parser.ioc_extractor` (PR#2568 by Sebastian Wagner).
```
any instructions for people who used it in the past? How should they update? I would recommend adding a link here to instructions on how to migrate...

sebix replied:
solution will be documented in NEWS.md, see the other remark.

aaronkaplan commented on CHANGELOG.md:
```diff
@@ -30,8 +26,6 @@
   - Log the downloaded size in bytes to ease troubleshooting (PR#2554 by Sebastian Wagner).
   - Fix import for Timeout exception preventing another exception (fixes #2555, PR#2556 by Sebastian Wagner).
 - Remove `intelmq.bots.collectors.twitter` as it uses an unmaintained library and does not work any more (fixes #2346, #2441, PR#2568 by Sebastian Wagner).
-- Renamed `intelmq.bots.parser.twitter` to `intelmq.bots.parser.ioc_extractor` (PR#2568 by Sebastian Wagner).
```
how to deal with the migration for existing users of the twitter bot?

sebix replied:
duplicate of previous comment

aaronkaplan commented on CHANGELOG.md:
```diff
@@ -61,6 +57,7 @@
 - Remove empty page tutorials/intelmq-manager (PR#2562 by Sebastian Wagner).
 
 ### Packaging
+- Packages for Ubuntu 24.04 (by Sebastian Wagner, fixes #2571).
```
Again, link to any instructions needed? Are there migration instructions needed for Ubuntu 22.x -> 24.04 ?

sebix replied:
Do we want to document how to do a distribution upgrade?
FWIW: https://documentation.ubuntu.com/server/how-to/software/upgrade-your-release/index.html

aaronkaplan replied:
Yes! We want to *link* to what people should read. Linking is good.

sebix replied:
Added the link to the news file

aaronkaplan commented on CHANGELOG.md:
```diff
@@ -70,11 +67,21 @@
+- Type error with SQL output bot's `prepare_values` returning list instead of tuple (#2255).
+- `intelmq_psql_initdb` does not work for SQLite (#2202).
+- intelmqsetup: should install a default state file (#2175).
+- Misp Expert - Crash if misp event already exist (#2170).
```
that seems bad

sebix replied:
And there's no bug fix since 3 years. That's nothing we can do about it in the release procedure.

aaronkaplan commented on NEWS.md:
```diff
@@ -10,26 +10,23 @@ This file lists all changes which have an affect on the administration of IntelM
-
-### Postgres databases
+#### Twitter Parser renaming
+The Twitter parser is renamed to *IoC Extractor Parser* (`intelmq.bots.parsers.ioc_extractor`).
```
see above... what should the user do?

sebix replied:
Will add a remark on the automatism.

sebix replied:
fixed

aaronkaplan commented on intelmq/lib/upgrades.py:
```diff
@@ -952,6 +953,27 @@ def v322_removed_feeds_and_bots(configuration, harmonization, dry_run, **kwargs)
+            found_twitter_collector.append(bot_id)
+
+    if found_twitter_collector:
+        message = f"Found discontinued Twitter collector bot: {', '.join(found_twitter_collector)}"
```
same comment. The user will see this (hopefully) and then ... what?

sebix replied:
Right, there's the instruction missing. Will add it.

sebix replied:
fixed

aaronkaplan reviewed (changes requested):
please yes, let's add guidance to people who do not know the documentation by heart... let's be --verbose for them. Let's add a link to the upgrade path. --verbose makes sense for humans.

sebix commented:
> please yes, let's add guidance to people who do not know the documentation by heart... let's be --verbose for them. Let's add a link to the upgrade path. --verbose makes sense for humans.

I've already added the command to execute

sebix committed:
REL: 3.4.0 Release

this commit contains the entries in changelog and NEWS
update the linux package documentation
adds upgrade scripts for removed/renamed bots
and sets the version number

aaronkaplan commented:
Thank you @sebix !!!! Sorry for my finickiness...

As agreed, @kamil-certat : I think it would be awesome if you could test it in staging and possibly PROD as well.. Thanks!!!

kamil-certat commented:
I've installed all three on staging and when nothing breaks, it will go to prod tomorrow

aaronkaplan commented:
cool! Thanks!

kamil-certat commented:
Some observations, but without any consequences for the incoming release:

1) the (old) intelmq 3.3.1 package in PyPI is broken: it surprisingly contains `certbund_rules` bot, but without all necessary dependencies, and the bot does not import them in a safety way => using IntelMQ installed from pip is broken.
2) I install IntelMQ through pip, not the deb packages (because it's more flexible). It looks that the "simple" installation works great on Debian 12, but Debian 11 has too old setuptools, and one have to do a little gymnastic. This should not have any consequences for DEB packages, and in general, it's better to use a virtualenv or pipx - I'm just not there yet with this setup :(

kamil-certat reviewed (approved):
I haven't seen any issues on my deployments.

aaronkaplan commented:
Thank you very much @sebix and @kamil-certat . Merging.
